pandas/core/arrays/numpy_.py

```diff
@@ -56,7 +55,7 @@ def numpy_dtype(self):
 
     @property
     def name(self):
-        return self._name
+        return self._dtype.name
```

Comment:
this seems like a worthwhile cleanup independent of other decisions/discussion in this PR

Reply:
This is actually not a "clean-up", but for performance (the specific purpose of the PR). From profiling, it appeared that getting the name of a numpy dtype is very costly:

```
In [8]: dtype = np.dtype("int64")

In [9]: %timeit dtype.name 
1.78 µs ± 6.75 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

(accessing `.array` for a numpy dtyped series (so creating a PandasArray) gives more or less the same time now, so doing this gave a 2x speed-up)